core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -416,55 +417,100 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     this.consumerConfig.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "roundrobin-group")
     this.consumerConfig.setProperty(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG, classOf[RoundRobinAssignor].getName)
 
-    val consumerCount = 10
-    val rrConsumers = Buffer[KafkaConsumer[Array[Byte], Array[Byte]]]()
-    for (i <- 0 until consumerCount) {
-      rrConsumers += new KafkaConsumer(this.consumerConfig)
-    }
-
     // create two new topics, total number of partitions must be greater than number of consumers
     val topic1 = "topic1"
     val topic2 = "topic2"
     val subscriptions = createTopicAndSendRecords(topic1, 5, 100) ++ createTopicAndSendRecords(topic2, 8, 100)
 
-    // all consumers subscribe to all the topics and start polling
+    // create a group of consumers, subscribe the consumers to all the topics and start polling
     // for the topic partition assignment
-    val consumerPollers = Buffer[ConsumerAssignmentPoller]()
-    for (consumer <- rrConsumers) {
-      assertEquals(0, consumer.assignment().size)
-      consumer.subscribe(List(topic1, topic2).asJava)
-      val poller = new ConsumerAssignmentPoller(consumer)
-      consumerPollers += poller
-      poller.start()
-    }
+    val (rrConsumers, consumerPollers) = createConsumerGroupAndWaitForAssignment(10, List(topic1, topic2), subscriptions)
 
-    TestUtils.waitUntilTrue(() => {
-      val assignments = Buffer[Set[TopicPartition]]()
-      consumerPollers.foreach(assignments += _.consumerAssignment())
-      isPartitionAssignmentValid(assignments, subscriptions)
-    }, s"Did not get valid initial assignment for partitions ${subscriptions.asJava}")
+    // add one more consumer and validate re-assignment
+    addConsumersToGroupAndWaitForGroupAssignment(1, rrConsumers, consumerPollers, List(topic1, topic2), subscriptions)
+
+    // done with pollers and consumers
+    for (poller <- consumerPollers)
+      poller.shutdown()
 
-    // add one more consumer
-    val newConsumer = new KafkaConsumer[Array[Byte], Array[Byte]](this.consumerConfig)
-    newConsumer.subscribe(List(topic1, topic2).asJava)
-    val newPoller = new ConsumerAssignmentPoller(newConsumer)
-    rrConsumers += newConsumer
-    consumerPollers += newPoller
-    newPoller.start()
+    for (consumer <- rrConsumers)
+      consumer.unsubscribe()
+  }
 
-    // wait until topics get re-assigned
-    TestUtils.waitUntilTrue(() => {
-      val assignments = Buffer[Set[TopicPartition]]()
-      consumerPollers.foreach(assignments += _.consumerAssignment())
-      isPartitionAssignmentValid(assignments, subscriptions)
-    }, s"Did not get valid assignment for partitions ${subscriptions.asJava} after we added one more consumer")
+  @Test
+  def testMultiConsumerAssignment() {
```

Comment:
Wouldn't it be better then to name it to testMultiConsumerDefaultAssignment, since this test re-uses policy set in BaseConsumerTest? What if at some point we change the defaults used in BaseConsumerTest?